api/v1alpha1/ndb_api_helpers.go

```diff
@@ -153,6 +162,115 @@ func GenerateProvisioningRequest(ctx context.Context, ndbclient *ndbclient.NDBCl
+			if errorThroughChecks != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_SOFTWARE] = matchedProfile
+			}
+		}
+
+		//Custom Compute Profile Check and overriding the default values
+		computeProfile := dbSpec.Instance.Profiles.Compute
+		if computeProfile == (Profile{}) {
+			log.Info("No enrichment for compute profiles as no custom profile received for Compute. Hence, proceeding with default OOB compute profile")
+		} else {
+			isValidProfile, matchedProfile, errorThroughChecks := performProfileAvailabilityCheck(ctx, genericProfiles, computeProfile, PROFILE_TYPE_COMPUTE)
+			if errorThroughChecks != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_COMPUTE] = matchedProfile
+			}
+		}
+
+		//Custom Network Profile Check and overriding the default values
+		networkProfile := dbSpec.Instance.Profiles.Network
+		if networkProfile == (Profile{}) {
+			log.Info("No enrichment for network profiles as no custom profile received for Network. Hence, proceeding with default OOB network profile")
+		} else {
+			isValidProfile, matchedProfile, errorThroughChecks := performProfileAvailabilityCheck(ctx, dbEngineSpecificProfiles, networkProfile, PROFILE_TYPE_NETWORK)
+			if errorThroughChecks != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_NETWORK] = matchedProfile
+			}
+		}
+
+		//Custom DbParam Profile Check and overriding the default values
+		dbParamProfile := dbSpec.Instance.Profiles.DbParam
+		if dbParamProfile == (Profile{}) {
+			log.Info("No enrichment for database parameter profiles as no custom profile received for it. Hence, proceeding with default OOB dbParam profile")
+		} else {
+			isValidProfile, matchedProfile, errorThroughChecks := performProfileAvailabilityCheck(ctx, dbEngineSpecificProfiles, dbParamProfile, PROFILE_TYPE_DATABASE_PARAMETER)
+			if err != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_DATABASE_PARAMETER] = matchedProfile
+			}
+		}
+	}
+	return
+}
+
+func performProfileAvailabilityCheck(ctx context.Context, profiles []ProfileResponse, customProfile Profile, profileType string) (isValidProfile bool, matchedProfile ProfileResponse, err error) {
+	log := ctrllog.FromContext(ctx)
+	isValidProfile = false
+	err = nil
+	matchedProfile = ProfileResponse{}
+	for _, eachProfile := range profiles {
+		if eachProfile.Type == profileType {
+			if (eachProfile.Id == customProfile.Id) && (eachProfile.LatestVersionId == customProfile.VersionId) {
+				log.Info("Profile Matched for custom profile", "profileType", profileType, "profileValue", eachProfile)
+				isValidProfile = true
+				matchedProfile = eachProfile
+				break
+			}
+		}
+	}
```

Comment:
Can we reuse the util/filter function here?

Reply:
This filter funtion has been removed and replaced by the filtering available through util package